aws/dfs/dfs.tf

```diff
@@ -0,0 +1,443 @@
+  type        = string
+  default     = ""
+}
+
+variable "ipv6_address_count" {
+  description = "A number of IPv6 addresses to associate with the primary network interface. Amazon EC2 chooses the IPv6 addresses from the range of your subnet."
+  type        = number
+  default     = null
+}
+
+
+
+variable "ipv6_addresses" {
+  description = "Specify one or more IPv6 addresses from the range of the subnet to associate with the primary network interface"
+  type        = list(string)
+  default     = null
+}
+
+
+variable "root_block_device" {
+  description = "Customize details about the root block device of the instance. See Block Devices below for details"
+  type        = list(map(string))
+  default     = []
+}
+
+variable "ebs_block_device" {
+  description = "Additional EBS block devices to attach to the instance"
+  type        = list(map(string))
+  default     = []
+}
+
+variable "ephemeral_block_device" {
+  description = "Customize Ephemeral (also known as Instance Store) volumes on the instance"
+  type        = list(map(string))
+  default     = []
+}
+
+variable "network_interface" {
+  description = "Customize network interfaces to be attached at instance boot time"
+  type        = list(map(string))
+  default     = []
+}
+
+
+variable "metadata_options" {
+  description = "Customize the metadata options of the instance"
+  type        = map(string)
+  default     = {}
+}
+
+data "aws_ami" "windows" {
+  most_recent = true
+  filter {
+    name   = "name"
+    values = ["Windows_Server-2016-English-Full-Base-*"]
+  }
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+  owners = ["801119661308"] # Canonical
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+# MODULES / RESOURCES
+# ----------------------------------------------------------------------------------------------------------------------
+
+#########################################################
+# Security group for DFS Namespace. All traffic from my IP.
+#########################################################
+
+variable ingress_cidrs {
+  type        = list(string)
+  default     = [""]
+  description = "description"
+}
+
+
+variable "dfs_tcp_allowed_ports" {
+  type = list(object({
+    from_port = number
+    to_port = number
+  }))
+}
+
+
+variable "dfs_udp_allowed_ports" {
+  type = list(object({
+    from_port = number
+    to_port = number
+  }))
+}
+
+
+
+# ----------------------------------------------
+####### Security Group 
+# ----------------------------------------------
+resource "aws_security_group" "default" {
+  count = var.create ? 1 : 0
+  name  = local.module_prefix
+  description = "common FSx ports"
+  vpc_id      = var.vpc_id
+  egress {
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Egress"
+    from_port   = 0
+    protocol    = "-1"
+    to_port     = 0
+  }
+  tags = merge(local.tags,map(

```

Comment:
- [ ] Remove MAP variables and use `var.tags` instead

Reply:
This gets into some odd logic. In the module there is two instances which each need a different "map-migrated" Value. So this is passed in as a list. Tags cannot be lists, so it errors out. This is an override to remove it from the the map. I agree this does not make the module friendly. I think we should look into how we can apply this extra tag outside of the module in a supplemental file to help clean this up and keep it generic+